From e67b31a6d61ec67c3cd89832a96e266f3470b5f0 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 25 Apr 2022 20:20:08 +0000
Subject: [PATCH] unbundle: add //services/data_decoder:lib to libxml_utils
 visibility

Also add a note to keep original and unbundle BUILD.gn files in sync.

Bug: 1306942
Change-Id: I83a57abfd7f2390232b1cf81060d79595f2aec55
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3488058
Reviewed-by: Lei Zhang <thestig@chromium.org>
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#995726}
---
 build/linux/unbundle/libxml.gn | 4 ++++
 third_party/libxml/BUILD.gn    | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/build/linux/unbundle/libxml.gn b/build/linux/unbundle/libxml.gn
index 243b62984fcb0..42f1aac867f49 100644
--- a/build/linux/unbundle/libxml.gn
+++ b/build/linux/unbundle/libxml.gn
@@ -12,6 +12,7 @@ source_set("libxml") {
   public_configs = [ ":system_libxml" ]
 }
 
+# Please keep in sync with //third_party/libxml/BUILD.gn.
 static_library("libxml_utils") {
   # Do not expand this visibility list without first consulting with the
   # Security Team.
@@ -19,6 +20,7 @@ static_library("libxml_utils") {
     ":xml_reader",
     ":xml_writer",
     "//base/test:test_support",
+    "//services/data_decoder:lib",
     "//services/data_decoder:xml_parser_fuzzer_deps",
   ]
   sources = [
@@ -28,6 +30,7 @@ static_library("libxml_utils") {
   public_configs = [ ":system_libxml" ]
 }
 
+# Please keep in sync with //third_party/libxml/BUILD.gn.
 static_library("xml_reader") {
   # Do not expand this visibility list without first consulting with the
   # Security Team.
@@ -44,6 +47,7 @@ static_library("xml_reader") {
   deps = [ ":libxml_utils" ]
 }
 
+# Please keep in sync with //third_party/libxml/BUILD.gn.
 static_library("xml_writer") {
   # The XmlWriter is considered safe to use from any target.
   visibility = [ "*" ]
diff --git a/third_party/libxml/BUILD.gn b/third_party/libxml/BUILD.gn
index 24688c1643c77..f408123aeb7f9 100644
--- a/third_party/libxml/BUILD.gn
+++ b/third_party/libxml/BUILD.gn
@@ -75,6 +75,7 @@ config("libxml_warnings") {
   }
 }
 
+# Please keep in sync with //build/linux/unbundle/libxml.gn.
 static_library("xml_reader") {
   # Do not expand this visibility list without first consulting with the
   # Security Team.
@@ -98,6 +99,7 @@ static_library("xml_reader") {
   configs += [ ":libxml_config" ]
 }
 
+# Please keep in sync with //build/linux/unbundle/libxml.gn.
 static_library("xml_writer") {
   # The XmlWriter is considered safe to use from any target.
   visibility = [ "*" ]
@@ -112,6 +114,7 @@ static_library("xml_writer") {
   configs += [ ":libxml_config" ]
 }
 
+# Please keep in sync with //build/linux/unbundle/libxml.gn.
 static_library("libxml_utils") {
   # Do not expand this visibility list without first consulting with the
   # Security Team.
-- 
2.35.1

